pkg/machines/components/vm/disks/diskAdd.jsx

```diff
@@ -379,6 +411,26 @@ export class AddDiskModalBody extends React.Component {
             this.setState({ busType: value });
             break;
         }
+        case 'file': {
+            stateDelta.file = value;
+            this.setState(prevState => {
+                if (value.endsWith(".iso"))
+                    this.onValueChanged("device", "cdrom");
```

Comment:
Hmm this maybe would worth a comment why you don't use stateDelta here.
As in: use onValueChange instead of setState in order to perform dependent state changes